### PR TITLE
Add production flag

### PR DESCRIPTION
**Pull request by faustinoaq**
Current settings are trying to install ameba (dev dependency) on production environments.

This PR solve that 👍

### CODE REVIEW

File: bin/compile [hunk 1]
```diff
@@ -74,7 +74,7 @@ if [ ! -e "$amber_executable" ]; then
       cd $BUILD_DIR/lib/amber
       sed -in "4d" $BUILD_DIR/lib/amber/src/amber/cli/commands/database.cr
       sed -in "68,70d" $BUILD_DIR/lib/amber/shard.yml
-      crystal deps update
+      crystal deps update --production
       mkdir -p $BUILD_DIR/bin
       echo -e "\n"
```
Comment (CaDs) [inline bin/compile]: I was wondering if —release would be a good thing to have here

File: bin/compile [hunk 2]
```diff
@@ -74,7 +74,7 @@ if [ ! -e "$amber_executable" ]; then
...
       crystal build --no-debug -o $BUILD_DIR/bin/amber $BUILD_DIR/lib/amber/src/amber/cli.cr
```
Comment (CaDs) [inline bin/compile]: I was wondering if —release would be a good thing to have here

File: bin/compile [hunk 3]
```diff
@@ -64,7 +64,7 @@ PATH="${PATH}:${CRYSTAL_DIR}/bin"
 start "Install project dependencies"
   cd $BUILD_DIR
   echo -e "\n"
-  crystal deps
+  crystal deps --production
```
Comment (veelenga) [inline bin/compile]: Please use `shards install --production` instead. See

* https://github.com/crystal-lang/crystal/issues/5510
* https://github.com/crystal-lang/crystal/pull/5544

Comment (faustinoaq) [reply bin/compile]: Done! 👍

File: bin/compile [hunk 4]
```diff
@@ -74,10 +74,10 @@ if [ ! -e "$amber_executable" ]; then
       cd $BUILD_DIR/lib/amber
       sed -in "4d" $BUILD_DIR/lib/amber/src/amber/cli/commands/database.cr
       sed -in "68,70d" $BUILD_DIR/lib/amber/shard.yml
```
Comment (faustinoaq) [inline bin/compile]: @eliasjpr Does this still work?

Comment (CaDs) [reply bin/compile]: @faustinoaq adding --production flag only didn't fix the deploy issue.

Commenting out `sed -in "68,70d" $BUILD_DIR/lib/amber/shard.yml` did

**Pull request merged**